### PR TITLE
Refactor naming of rules and files

### LINKED ISSUE #18: Refactor folder structure to be package-oriented
## Package Names

First, we need to decide on the terminology. The table below shows the terminology for this the various packages we support.

| Package Management System | Package Format | Package Repository/Host |
|--------	|--------	|-----------------	|
| Maven 	| JAR 	| Maven Central 	|
| PIP 	| ? 		| PyPI 			|
| NPM 	| ? 		| NPM 			|
| APT 	| DEB 	| (decentralised) 	|
| RPM 	| RPM 	| (decentralised) 	|
| Brew	| ?		| Brew			|
| GitHub  | ?            | GitHub                 |

We need to fill in the `?` above, and Docker is also an anomaly here because it's not really a "package manager". However, I think we should stick with a consistent style of naming. In this case, let's go with the first column ("Package Manager") as the canonical names. E.g. don't use "deb" as a package name, nor should you use "pypi".  Wherever the package format is expected, we should use the second column ("Package Format").

## Component Names

We need to determine the components of "distribution", which is ultimately comprised of two steps:
- `assembly`: the task of "building" the distribution
- `deployment`: the task of "uploading" the distribution

If one of the packages do not have one of the steps, e.g. no `assemble` step, then that component can just be omitted. 

---

Project Structure:
```
common
  - assemble_targz
  - assemble_zip
apt
  - assemble_apt
  - deploy_apt
brew
  - deploy_brew
github
  - deploy_github
maven
  - assemble_maven
  - deploy_maven
npm
  - assemble_npm
  - deploy_npm
pip
  - assemble_pip
  - deploy_pip
rpm
  - assemble_rpm
  - deploy_rpm
```

### LINKED ISSUE #27: Rename distribution_structure to distribution_src and think of a better name for output_package_name
TODO: fill in details

### PR DESCRIPTION
**Pull request by vmax**
## What is the goal of this PR?

Achieve clearer definitions for rules
Fixes #18
Closes #27 as no longer relevant

## What are the changes implemented in this PR?

Implemented the following rules:
* `assemble_apt` and `deploy_apt` in `apt/rules.bzl`
* `assemble_rpm` and `deploy_rpm` in `rpm/rules.bzl`
* `assemble_targz`, `java_deps` and `assemble_zip` in `common`

### CODE REVIEW

File: common/rules.bzl [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        progress_message = "Generating tarball with Java deps: {}".format(
+            ctx.outputs.distribution.short_path)
+    )
+
+
+java_deps = rule(
+    attrs = {
+        "target": attr.label(mandatory=True),
+        "java_deps_root": attr.string(
+            default = "services/lib/",
```
[the remark this answers is not in the document]
Comment (vmax) [reply common/rules.bzl]: @lolski I can agree here

File: common/rules.bzl [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            doc = "Folder inside archive to put JARs into"
+        ),
+        "_java_deps_builder": attr.label(
+            default = "//common:java_deps",
```
[the remark this answers is not in the document]
Comment (vmax) [reply common/rules.bzl]: @lolski attributes starting with `_` are private and **must** have default value

File: common/rules.bzl (hunk 1 above)
[the remark this answers is not in the document]
Comment (lolski) [reply common/rules.bzl]: @vmax can you add this comment to the issue I just created?

File: common/rules.bzl (hunk 2 above)
[the remark this answers is not in the document]
Comment (lolski) [reply common/rules.bzl]: Same here, please add this comment to the issue.

**Pull request merged**